core/common/src/utils.rs

```diff
@@ -105,9 +108,14 @@ pub async fn discover_service_using_chariott(
     expected_communication_kind: &str,
     expected_communication_reference: &str,
 ) -> Result<String, Status> {
-    let mut client = ServiceRegistryClient::connect(chariott_uri.to_string())
-        .await
-        .map_err(|e| Status::internal(e.to_string()))?;
+    let uri = get_uri(chariott_uri).map_err(|err| {
+        Status::failed_precondition(format!(
+            "Unable to get container environment variable with error: {err}"
```

Comment:
Perhaps include the name of the environemnt variable?

Reply:
Updated to add the env variable name in get_uri.